### PR TITLE
feat: add dashboard widget colspan support

### PR DESCRIPTION
**Pull request by ugur-vaadin**
## Description

Adds colspan support to dashboard widget.

Added API:
- `int getColspan()`: Returns the colspan of the widget
- `void setColspan(int colspan)`: Sets the colspan of the widget

Fixes https://github.com/orgs/vaadin/projects/70/views/1?pane=issue&itemId=76144368

Part of https://github.com/vaadin/platform/issues/6626

## Type of change

Feature

### CODE REVIEW

File: vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java [hunk 1]
```diff
@@ -159,7 +159,7 @@ public Integer getMaximumColumnCount() {
     /**
      * Sets the maximum column count of the dashboard.
      *
-     * @param maxCount
+     * @param maxColCount
```
Comment (ugur-vaadin) [inline vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java:162]: Typo fix from the previous PR, included into this one.

File: vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java [hunk 2]
```diff
@@ -244,6 +244,7 @@ private JsonArray createItemsJsonArray() {
         for (DashboardWidget widget : widgets) {
             JsonObject jsonItem = Json.createObject();
             jsonItem.put("nodeid", getWidgetNodeId(widget));
+            jsonItem.put("colspan", widget.getColspan());
```
Comment (tomivirkki) [inline vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java:247]: This isn't tested. Should there be a `getColspan()` in `DashboardWidgetElement`?

Comment (ugur-vaadin) [reply vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java:247]: There are unit tests for this in `DashboardTest`, which checks the `colspan` through the `items` set to the client. That's why I didn't add any ITs. Do you think we should still add ITs?

`getColspan()` in `DashboardWidgetElement` makes sense. I will add that.

Comment (ugur-vaadin) [reply vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java:247]: Added `getColspan()` to `DashboardWidgetElement`. It is quite ugly, but it works. I couldn't see a better way to implement this. Any suggestions are welcome.

Comment (tomivirkki) [reply vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java:247]: No need to access the wrapper if you use `getComputedStyle`:

```java
public Integer getColspan() {
    var colspanStr = getComputedCssValue("--vaadin-dashboard-item-colspan");
    return colspanStr.isEmpty() ? null : Integer.valueOf(colspanStr);
}

private String getComputedCssValue(String propertyName) {
    return (String) executeScript(
            "return getComputedStyle(arguments[0]).getPropertyValue(arguments[1]);",
            this, propertyName);
}
```

Comment (tomivirkki) [reply vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java:247]: > There are unit tests for this in DashboardTest, which check the colspan through the items sent to the client.

Ah, somehow I didn't catch these tests failing with the line removed. All good regarding testing the component logic then.
However, nothing is currently testing the `getColspan()` API of `DashboardWidgetElement`.

Comment (ugur-vaadin) [reply vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java:247]: Thanks, I applied your computed style suggestion to get the colspan. For `getColspan()` being untested, I was not sure if we should add anything to test it. But I added 2 simple ITs that should cover it.

File: vaadin-dashboard-flow-parent/vaadin-dashboard-testbench/src/main/java/com/vaadin/flow/component/dashboard/testbench/DashboardWidgetElement.java [hunk 3]
```diff
@@ -25,4 +25,21 @@ public class DashboardWidgetElement extends TestBenchElement {
     public String getTitle() {
         return getPropertyString("widgetTitle");
     }
+
+    /**
+     * Returns the colspan of the widget.
+     *
+     * @return the {@code --vaadin-dashboard-item-colspan} style from the
+     *         wrapper of the web component
```
Comment (tomivirkki) [inline vaadin-dashboard-flow-parent/vaadin-dashboard-testbench/src/main/java/com/vaadin/flow/component/dashboard/testbench/DashboardWidgetElement.java:33]: The new implementation isn't getting the value from the wrapper.

Comment (ugur-vaadin) [reply vaadin-dashboard-flow-parent/vaadin-dashboard-testbench/src/main/java/com/vaadin/flow/component/dashboard/testbench/DashboardWidgetElement.java]: Updated the wording.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**